### PR TITLE
added user choice for difficulty level with test

### PR DESCRIPTION
**Pull request by jennatuckerdeveloper**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -121,6 +122,18 @@ class App extends Component {
     }
   }
 
+  handleDifficulty (e) {
+    if (e.keyCode === RETURN || e.keyCode === SPACEBAR) {
+      const userChoice = parseInt(e.target.value, 10)
+      if (userChoice === 1 || userChoice === 2 || userChoice === 3) {
```
Comment (psbanka) [inline src/App.js]: `if ([1, 2, 3].includes(userChoice)) ... `

File: src/App.js [hunk 2]
```diff
@@ -121,6 +122,18 @@ class App extends Component {
...
+        const gameObject = this.state.game
+        gameObject['difficulty'] = userChoice
+        gameObject['gameState'] = NAMING
```
Comment (psbanka) [inline src/App.js]: You're mutating state!
```
const gameObject = Object.assign({}, this.state.game)
...
```

File: src/DifficultyPage.test.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        handleDifficulty={() => { handleDifficultyTriggered = true }}
+      />
+    )
+  })
+
+  it('triggers handleDifficulty when user inputs difficulty level', () => {
+    const difficultyInput = wrapper.find('#difficultyInput')
+    const e = {}
+    difficultyInput.simulate('keyDown', e)
+    expect(handleDifficultyTriggered).toBe(true)
```
Comment (psbanka) [inline src/DifficultyPage.test.js:30]: 👍 good test.

File: src/Integration.test.js [hunk 4]
```diff
@@ -17,12 +17,39 @@ describe('integration testing', () => {
...
+    const enterDifficulty = app.find('#difficultyInput')
+    const mockDifficutlyEntry = {keyCode: 13, target: {value: '1'}}
+    enterDifficulty.simulate('keyDown', mockDifficutlyEntry)
+  }
+
+  const toPlaying = function () {
+    const confirmation = app.find('#confirmNames')
+    const mockOnConfirmEvent = {target: {value: 'y'}}
+    confirmation.simulate('keyDown', mockOnConfirmEvent)
+  }
```
Comment (psbanka) [inline src/Integration.test.js:30]: good utility functions 😄

**Review by psbanka — CHANGES_REQUESTED** [review]
Everything's fine except the state-mutation thing.

**Comment by jennatuckerdeveloper** [comment]
@psbanka Changes made.  Thank you!

**Review by psbanka — APPROVED** [review]

**Pull request merged**